### PR TITLE
clarify archeio docs

### PR DESCRIPTION
**Pull request by BenTheElder**
- We should discuss the current design before the original design
- Clarifying that you *must* join the mailinglist to access the original design sketch Google doc, still getting a lot of requests for this doc and this is the only place we currently link it AFAIK. Also clarifying that it's *not* current and the design has evolved, and outlining how that has evolved a bit.
- Adding another pointer to the main readme and particularly the stability section
- Clarifying repeatedly that while we document for transparency, you cannot rely on the implementation details
- Adding a bit more context re: security posture

I need to do a follow-up updating the "request handling" diagrams, we've made some slight changes since the last update.

### CODE REVIEW

**Comment by BenTheElder** [comment]
I think actually the only thing missing from request handling doc is `/v2/_catalog`, we did update it when doing #147

File: cmd/archeio/README.md [hunk 1]
```diff
@@ -31,20 +25,60 @@ For a rough TLDR of the current design:
 - Mirrors *of content* blobs are hosted in S3 buckets in AWS
 - AWS clients are detected by client IP address and redirect to a local S3 bucket copy
 *only* when requesting content blobs, *not* manifests, manifest lists, tags etc.
-- All other requests are redirected to the original upstream registry
+- GCP clients and other requests are rerouted to the regional upstream source ARtifact Registries
```
Comment (dims) [inline cmd/archeio/README.md:28]: lowercase

**Pull request merged**